lib/elixir/lib/gen_server.ex

```diff
@@ -785,8 +785,20 @@ defmodule GenServer do
             {_, name} -> name
           end
 
-        pattern = '~p ~p received unexpected message in handle_info/2: ~p~n'
-        :error_logger.error_msg(pattern, [__MODULE__, proc, msg])
+        :logger.error(
+          %{
+            label: {GenServer, :no_handle_info},
+            module: __MODULE__,
+            message: msg,
+            name: proc
+          },
+          %{
+            domain: [:otp, :elixir],
+            error_logger: %{tag: :error_msg},
+            report_cb: &GenServer.format_log/1
```

Comment:
Is there any reason why we don't pass this as `&format_log/1` and keep `format_log/1` private?

Reply:
Not really. I do not know why I did it that way.